### ISSUE TITLE
Apollo boost can't send cookies during cross-origin request

### ISSUE THREAD

**Issue by andriy-f** [apollographql/apollo-client#3407]
**Intended outcome:**

Cookies would be send during cross-origin request

**Actual outcome:**

Cookies were not sent during cross-origin request

**How to reproduce the issue:**

Make server on different domain which returns headers
```
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: <url of site with apollo-boost>
```

```
import ApolloClient from 'apollo-boost';

const client = new ApolloClient({ 
  uri: '<server on different domain>' ,
  fetchOptions: {
    credentials: 'include'
  }
});
```

**REASON:**
https://github.com/apollographql/apollo-client/blob/774f16fb3c80a247f072dc1368c85476d05b0f0b/packages/apollo-boost/src/index.ts#L77

(same-origin is hard-coded and not overridable

**Version**

- apollo-boost@0.1.6

**Comment by alexgorbatchev** [comment]
duplicates #3264

**Comment by hwillson** [comment]
Fixed in `apollo-client` 2.3.3. Thanks!
